### PR TITLE
DTSSTCI-353 and DTSSTCI-450

### PR DESCRIPTION
**Pull request by gmmagruder**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSSTCI-450

### Change description ###
Fix document upload and deletion functionality, update update/delete error and improve unit tests.
**NOTE:** This PR was branched off DTSSTCI-353: https://github.com/hmcts/sptribs-dss-update-case-web/pull/35

Screenshot 2024-03-04 at 15 23 33
Screenshot 2024-03-04 at 15 22 45
Screenshot 2024-03-05 at 09 34 20

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/app/controller/AppRequest.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export interface AppSession extends Session {
   lang: string | undefined;
   errors: FormError[] | undefined;
   addresses: [];
+  fileErrors: any[];
```
Comment (gmmagruder) [inline src/main/app/controller/AppRequest.ts:27]: Unchanged from DTSSTCI-535

File: src/main/app/controller/GetController.ts [hunk 2]
```diff
@@ -68,12 +68,16 @@ export class GetController {
...
+        fileErrors: req.session.hasOwnProperty('fileErrors') ? req.session.fileErrors : [],
         caseId: req.session.userCase?.id,
       };
 
       if (req.session?.errors) {
         req.session.errors = undefined;
       }
+      if (req.session?.fileErrors) {
+        req.session.fileErrors = [];
+      }
```
Comment (gmmagruder) [inline src/main/app/controller/GetController.ts:80]: Unchanged from DTSSTCI-535

File: src/main/steps/upload-documents/content.test.ts [hunk 3]
```diff
@@ -1,47 +1,49 @@
...
-      fileFormat: `The file you uploaded is in the wrong format.
-              Upload your file again in the correct format`,
+      noInput: 'You cannot continue without providing additional information or a document',
+      fileSize: `File size exceeds the maximum permitted value. Please upload a file that is less than 20MB (documents) or less than 30MB (multimedia files)`,
+      fileFormat: `This service only accepts files in the formats - Ms Word, MS Excel, PDF, JPG, PNG, TXT, RTF, MP4, MP3`,
       selectFileToUpload: 'Select a file to upload',
-      uploadError: 'The selected file could not be uploaded – try again',
-      maxFileError: 'You can only select up to 10 files at the same time',
-      fileDescriptionRequired: 'Enter a description for the file selected for upload',
+      uploadDeleteError: 'Document upload or deletion has failed. Please try again',
```
Comment (gmmagruder) [inline src/main/steps/upload-documents/content.test.ts:45]: Updated upload/delete error - copied from dss submit case

File: config/default.yaml [hunk 4]
```diff
@@ -21,8 +21,8 @@ api:
   serviceAuth: RPE_API
 uploadPolicy:
   documentSize: 20
-  multimediaSize: 500
-  maxNoOfFiles: 10
+  multimediaSize: 30
```
Comment (tomxelliott) [inline config/default.yaml:24]: This needs to be 500mb

File: src/main/app/controller/AppRequest.ts (hunk 1 above)
Comment (tomxelliott) [reply src/main/app/controller/AppRequest.ts:27]: How is this related to https://tools.hmcts.net/jira/browse/DTSSTCI-535?

File: src/main/app/controller/GetController.ts (hunk 2 above)
Comment (tomxelliott) [reply src/main/app/controller/GetController.ts:80]: Same as above

File: src/main/app/controller/AppRequest.ts (hunk 1 above)
Comment (gmmagruder) [reply src/main/app/controller/AppRequest.ts:27]: Sorry my mistake 353.

File: src/main/app/fileUpload/documentManager.ts [hunk 5]
```diff
@@ -1,39 +1,50 @@
...
-    data: {},
-    // httpsAgent: new https.Agent({ rejectUnauthorized: false }),
+export const deleteDocument = async (s2sToken, documentID, req) => {
+  const CASE_API_URL: string = config.get('services.sptribs.url')
+  const deleteUrl = `/doc/dss-orchestration/${documentID}/delete`;
+  const headers = {
+    authorization: `Bearer ${req.session.user.accessToken}`,
+    serviceAuthorization: s2sToken,
+  };
+  const DOCUMENT_DELETEMANAGER: AxiosInstance = axios.create({
```
Comment (tomxelliott) [inline src/main/app/fileUpload/documentManager.ts:35]: Do we need 2 instances of Axios?

Comment (tomxelliott) [reply src/main/app/fileUpload/documentManager.ts:35]: One should be able to handle upload and delete?

File: src/main/steps/upload-documents/content.test.ts [hunk 6]
```diff
@@ -1,47 +1,49 @@
-import { cy, en, form, generateContent } from './content';
+import { cy, en, form, generateContent, getErrors } from './content';
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 const englishContent = () => ({
```
Comment (tomxelliott) [inline src/main/steps/upload-documents/content.test.ts:4]: can we change this to en and Welsh one to cy to keep consistency?

File: src/main/steps/upload-documents/content.test.ts [hunk 7]
```diff
@@ -54,47 +56,48 @@ const englishContent = () => ({
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 const welshContent = () => ({
```
Comment (tomxelliott) [inline src/main/steps/upload-documents/content.test.ts:58]: can this be cy please

**Review by tomxelliott — APPROVED** [review]

**Review by tomxelliott — APPROVED** [review]

File: src/main/steps/upload-documents/content.test.ts [hunk 8]
```diff
@@ -1,47 +1,49 @@
...
     documentUpload: {
-      noDocumentUploaded: 'Please choose a file.',
-      fileSize: `The file you uploaded is too large.
-              Maximum file size allowed is 20MB`,
-      multimediaFileSize: 'The file you uploaded is too large. Maximum file size allowed is 500MB for multimedia files',
-      fileFormat: `The file you uploaded is in the wrong format.
-              Upload your file again in the correct format`,
+      noInput: 'You cannot continue without providing additional information or a document',
+      fileSize: `File size exceeds the maximum permitted value. Please upload a file that is less than 20MB (documents) or less than 500MB (multimedia files)`,
+      fileFormat: `This service only accepts files in the formats - Ms Word, MS Excel, PDF, JPG, PNG, TXT, RTF, MP4, MP3`,
```
Comment (oleon23) [inline src/main/steps/upload-documents/content.test.ts:43]: For consistency should it be `MS Word, MS Excel` as in the list above L22

File: src/main/steps/upload-documents/content.test.ts [hunk 9]
```diff
@@ -54,47 +56,48 @@ const englishContent = () => ({
...
-      noDocumentUploaded: 'Please choose a file. - welsh',
-      fileSize: `The file you uploaded is too large.
-              Maximum file size allowed is 20MB - welsh`,
-      multimediaFileSize:
-        'The file you uploaded is too large. Maximum file size allowed is 500MB for multimedia files - welsh',
-      fileFormat: `The file you uploaded is in the wrong format.
-              Upload your file again in the correct format - welsh`,
+      noInput: 'You cannot continue without providing additional information or a document - welsh',
+      fileSize: `File size exceeds the maximum permitted value. Please upload a file that is less than 20MB (documents) or less than 500MB (multimedia files) - welsh`,
+      fileFormat: `This service only accepts files in the formats - Ms Word, MS Excel, PDF, JPG, PNG, TXT, RTF, MP4, MP3 - welsh`,
```
Comment (oleon23) [inline src/main/steps/upload-documents/content.test.ts:97]: Same as above `Ms Word` -> `MS Word`

File: src/main/steps/upload-documents/content.ts [hunk 10]
```diff
@@ -55,47 +57,48 @@ export const en = () => ({
...
-      noDocumentUploaded: 'Please choose a file. - welsh',
-      fileSize: `The file you uploaded is too large.
-            Maximum file size allowed is 20MB - welsh`,
-      multimediaFileSize:
-        'The file you uploaded is too large. Maximum file size allowed is 500MB for multimedia files - welsh',
-      fileFormat: `The file you uploaded is in the wrong format.
-            Upload your file again in the correct format - welsh`,
+      noInput: 'You cannot continue without providing additional information or a document - welsh',
+      fileSize: `File size exceeds the maximum permitted value. Please upload a file that is less than 20MB (documents) or less than 500MB (multimedia files) - welsh`,
+      fileFormat: `This service only accepts files in the formats - Ms Word, MS Excel, PDF, JPG, PNG, TXT, RTF, MP4, MP3 - welsh`,
```
Comment (oleon23) [inline src/main/steps/upload-documents/content.ts:98]: Same as above `Ms Word` -> `MS Word`

File: src/main/steps/upload-documents/postController.test.ts [hunk 11]
```diff
@@ -149,35 +191,97 @@ describe('Testing the post controller', () => {
...
+    expect(req.session?.fileErrors[0].href).toEqual('#file-upload-1');
+  });
+
+  test('uploadFileError fileFormat', () => {
+    const newRequest = req;
+    newRequest.session['save'] = () => '';
+    controller.uploadFileError(newRequest, res, '', 'fileFormat');
+    expect(res.redirect).not.toHaveBeenCalled();
+    expect(req.session?.fileErrors).toHaveLength(1);
+    expect(req.session?.fileErrors[0].text).toEqual('This service only accepts files in the formats - Ms Word, MS Excel, PDF, JPG, PNG, TXT, RTF, MP4, MP3');
```
Comment (oleon23) [inline src/main/steps/upload-documents/postController.test.ts:270]: Will need to be updated as above also.

File: src/main/steps/upload-documents/postController.test.ts [hunk 12]
```diff
@@ -58,44 +57,71 @@ describe('Testing the post controller', () => {
...
+    [
+      'sample.jpg',
+      'sample.jpeg',
+      'sample.bmp',
+      'sample.png',
+      'sample.pdf',
+      'sample.doc',
+      'sample.docx',
+      'sample.rtf',
+      'sample.xlsx',
```
Comment (oleon23) [inline src/main/steps/upload-documents/postController.test.ts:73]: Should we also include `.xls` here?

File: src/main/steps/upload-documents/content.test.ts (hunk 8 above)
Comment (oleon23) [reply src/main/steps/upload-documents/content.test.ts:43]: Should we also list out BMP in the list as it is valid?

Comment (gmmagruder) [reply src/main/steps/upload-documents/content.test.ts:43]: Error messaging is consistent with dss submit case. Happy to ask if this should be changed.

File: src/main/steps/upload-documents/postController.test.ts (hunk 12 above)
Comment (gmmagruder) [reply src/main/steps/upload-documents/postController.test.ts:73]: Good spot, thanks.

**Review by oleon23 — DISMISSED** [review]
Happy to approve and spelling can be updated later in both services as needed.

**Review by rajaram-nagarajan — APPROVED** [review]

**Review by tomxelliott — DISMISSED** [review]

**Review by tomxelliott — DISMISSED** [review]

**Review by Jefex7 — DISMISSED** [review]

**Review by oleon23 — DISMISSED** [review]

**Review by oleon23 — DISMISSED** [review]

**Review by rajaram-nagarajan — DISMISSED** [review]
LGTM

File: src/main/steps/upload-documents/template.njk [hunk 13]
```diff
@@ -64,21 +64,15 @@
...
-         {% set documentId = document.documentId %}
-          <dd class="govuk-summary-list__actions">
-              <a class="govuk-link" href="{{postURL}}?removeId={{documentId}}">
-                {{delete}}
-                </a>
-          </dd>
-      </div>
+      <li class="uploadedFile govuk-!-padding-top-2 govuk-!-padding-bottom-3 govuk-section-break govuk-section-break--visible">
+        {{ document.fileName }}
+        {% set documentId = document.documentId%}
```
Comment (tomxelliott) [inline src/main/steps/upload-documents/template.njk:71]: for sake of consistency could we have a space between Id and %}

**Review by tomxelliott — DISMISSED** [review]

**Review by rajaram-nagarajan — DISMISSED** [review]

**Review by tomxelliott — DISMISSED** [review]

**Review by tomxelliott — DISMISSED** [review]

**Review by Jefex7 — DISMISSED** [review]

**Review by tomxelliott — APPROVED** [review]

**Review by Jefex7 — APPROVED** [review]

**Pull request merged**